tests/flow/test_bidirectional_traversals.py

```diff
@@ -310,3 +310,33 @@ def test11_bidirectional_multiple_edge_type(self):
                            ['d', 'c']]
 
         self.env.assertEquals(actual_result.result_set, expected_result)
+
+    # Test bidirectional traversals resolved by an ExpandInto op.
+    def test12_bidirectional_expand_into(self):
+        query = """MATCH (a), (b) WITH a, b MATCH (a)-[e:E]-(b) RETURN a.val, b.val ORDER BY a.val, b.val"""
+        actual_result = acyclic_graph.query(query)
+        # Each relation should appear twice with the source and destination swapped in the second result.
+        expected_result = [['v1', 'v2'],
+                           ['v2', 'v1'],
+                           ['v2', 'v3'],
+                           ['v3', 'v2']]
+        self.env.assertEquals(actual_result.result_set, expected_result)
+
+        # Verify result against the equivalent conditional traversal.
+        query = """MATCH (a)-[:E]-(b) RETURN a.val, b.val ORDER BY a.val, b.val"""
+        traverse_result = acyclic_graph.query(query)
+        self.env.assertEquals(actual_result.result_set, traverse_result.result_set)
+
+        # Test undirected traversals with a referenced edge.
+        query = """MATCH (a), (b) WITH a, b MATCH (a)-[e:E]-(b) RETURN ID(e), a.val, b.val ORDER BY ID(e), a.val, b.val"""
+        actual_result = acyclic_graph.query(query)
+        expected_result = [[0, 'v1', 'v2'],
+                           [0, 'v2', 'v1'],
+                           [1, 'v2', 'v3'],
+                           [1, 'v3', 'v2']]
+        self.env.assertEquals(actual_result.result_set, expected_result)
+
+        # Verify result against the equivalent conditional traversal.
+        query = """MATCH (a)-[e:E]-(b) RETURN ID(e), a.val, b.val ORDER BY ID(e), a.val, b.val"""
```

Comment:
set `id` as edge property as we can't rely on the order of the `redisgraph-py` flushes the data